griptape/engines/summary/base_summary_engine.py

```diff
@@ -2,12 +2,11 @@
 from attr import define
 from griptape.artifacts import TextArtifact, BaseArtifact
 
-
 @define
 class BaseSummaryEngine(ABC):
-    def summarize_text(self, text: str) -> str:
-        return self.summarize_artifacts([TextArtifact(text)]).value
+    def summarize_text(self, text: str, length: int = None, target_audience: str = None, format: str = None) -> str:
```

Comment:
Those should be class properties for:
- DRYness across class methods and
- ease of integration with tasks and `TextMemoryActivitiesMixin.summarize`.

Reply:
Additionally, we always use `Optional`s in type hints if the default value is `None`. In this case it should be `Optional[int]`, `Optional[str]`, and `Optional[str]`.